### PR TITLE
Adding hardware security modules RP with dedicated HSM swagger

### PR DESCRIPTION
**Pull request by schaabs**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
@schaabs examples?

**Comment by schaabs** [comment]
@jhendrixMSFT I'll add examples shortly

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+        }
+      }
+    },
+    "Sku": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "SKU of the dedicated HSM",
+          "enum": [
+            "SafeNet Network HSM 7"
```
Comment (jkapil) [inline specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json]: This enum value doesn't match the code. Please sync with Nitin Goel

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+          "description": "The list of dedicated HSM resources."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of dedicated HSM resources."
+        }
+      },
+      "description": "List of dedicated HSM resources."
+    },
+    "KeyVaultError": {
```
Comment (jkapil) [inline specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json]: Is the usage of KeyVaultError intentional? Could it be say DedicatedHsmError

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json (hunk 1 above)
Comment (ngoel110) [reply specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json]: Please use this: SafeNet Luna Network HSM A790

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json (hunk 2 above)
Comment (ngoel110) [reply specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json]: Yes, please use DedicatedHsmError. It was my fault.

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/examples/DedicatedHsm_CreateOrUpdate.json [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        "api-version": "2018-10-31-preview",
+        "parameters": {
+            "properties": {
+                "networkProfile": {
+                    "subnet": {
+                        "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/hsm-group/providers/Microsoft.Network/virtualNetworks/stamp0/subnets/stamp0"
+                    },
+                    "networkInterfaces": [
+                        {
+                            "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/hsm-group/providers/Microsoft.Network/networkInterfaces/hsm_vnic",
```
Comment (ngoel110) [inline specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/examples/DedicatedHsm_CreateOrUpdate.json]: networkInterfaces id is read only field which is set by RP

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/examples/DedicatedHsm_CreateOrUpdate.json [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                            "privateIpAddress": "1.0.0.1"
+                        }
+                    ]
+                },
+                "stampId": "0"
```
Comment (ngoel110) [inline specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/examples/DedicatedHsm_CreateOrUpdate.json]: It has to be one of these "stamp01"\"stamp02"\stamp03

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json [hunk 5]
```diff
@@ -0,0 +1,631 @@
...
+        }
+      }
+    },
+    "DedicatedHsmProperties": {
+      "properties": {
+        "networkProfile": {
+          "$ref": "#/definitions/NetworkProfile",
+          "description": "Specifies the network interfaces of the dedicated hsm."
+        },
+        "stampId": {
```
Comment (ravbhatnagar) [inline specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json]: readonly

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json [hunk 6]
```diff
@@ -0,0 +1,631 @@
...
+          "type": "string",
+          "description": "This field will be used when RP does not support Availability zones."
+        },
+        "statusMessage": {
```
Comment (ravbhatnagar) [inline specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json]: how is this used? is it status during provisioning time?

**Review by ravbhatnagar — COMMENTED** [review]
Looks good. Couple of minor things.

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json (hunk 5 above)
Comment (schaabs) [reply specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2018-10-31-preview/dedicatedhsm.json]: stampId is not read-only it is settable at create time.

**Comment by jhendrixMSFT** [comment]
Waiting for confirmation that the service has been deployed.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
OK sorry I missed that there are model validation failures (we switched up how some of this stuff is reported and I overlooked it).  See the [log](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/410560045).

**Comment by schaabs** [comment]
@jhendrixMSFT all the model validation errors have been fixed.  The service deployment to prod is still pending. @ngoel110 will reply back to the PR when it can be merged.

**Comment by schaabs** [comment]
@ngoel110 has the service been deployed to PROD yet?

**Comment by ngoel110** [comment]
@schaabs , @jhendrixMSFT , it is deployed but not registered with ARM. Will update when it is available

**Comment by ngoel110** [comment]
RP is deployed in West Central US, centralUsEuap, Eastus2Euap. Please merge the changes.

**Review by jhendrixMSFT — APPROVED** [review]
OK sorry I missed that there are model validation failures (we switched up how some of this stuff is reported and I overlooked it).  See the [log](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/410560045).

**Pull request merged**